kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -401,33 +404,35 @@ define([
-                var metadataFile = JSON.parse(dataString);
-                // these files are always a single line, so the head will contain the contents.
-                // but we parse it out and re-stringify it so it's pretty.
-                // XXX - while doing this, I ran into a NaN issue in the file, specifically on the key illumina_read_insert_size_avg_insert.
-                //       So we nuke any NaN fields to make it valid again.
-                var metadataJSON = JSON.parse(metadataFile.head.replace(/NaN/g, '\"\"'));
-                var metadataContents = JSON.stringify(metadataJSON, null, 2)
-
-                $tabs.addTab(
-                  {
-                    tab : 'Metadata',
-                    content : $.jqElem('div')
-                      .addClass('kb-data-staging-metadata-file-lines')
-                      .append( metadataContents )
-                  }
-                );
-              })
-              .fail(function(xhr) {
-                // Don't actually need to do anything here - we assume that if it failed, it was due to the metadata file not existing. Yes, we generate
-                // a lot of messy extra metadata calls here since it's the only way to know if there's metadata is to look.
-              });
-            }
+            // attempt to load up a jgi metadata file, via the jgi-metadata endpoint. It'll only succeed if a jgi metadata file exists
+            // We can't do it in parallel, since if the metadata file doesn't exist the promise wouldn't properly complete.
+
+            self.stagingServiceClient.jgi_metadata({ path : filePath }).then( function(dataString, status, xhr) {
+              // XXX - while doing this, I ran into a NaN issue in the file, specifically on the key illumina_read_insert_size_avg_insert.
+              //       So we nuke any NaN fields to make it valid again.
+              var metadataJSON      = JSON.parse(dataString.replace(/NaN/g, '\"\"'));
+              var metadataContents  = JSON.stringify(metadataJSON, null, 2)
+
+              $tabs.addTab(
+                {
+                  tab : 'JGI Metadata',
+                  content : $.jqElem('div')
+                    .addClass('kb-data-staging-metadata-file-lines')
+                    .append( metadataContents )
+                }
+              );
+
+              // finally, empty and append the tabs container.
+              $tabsDiv.empty();
+              $tabsDiv.append($tabsContainer);
+            })
+            .fail(function (xhr) {
+              // if we failed, then there's no JGI metadata. That's fine. We still want to empty and append the tabs container.
+              // yes, yes, I could abstract this out into a separate function call so the code is DRY and not WET, but it's a duplicate
+              // of two lines that are right next to each other. I felt this was easier.
```

Comment:
You could also put those in a `.always` or `.finally` (I forget if this is a Promise or a jQuery deferred...) so it's always done at the end.

But it is nice to have a separate, explicit `.fail` block.